### PR TITLE
Reference guide complete for 19.50 release

### PR DESCRIPTION
**Pull request by ErikMHummel**
The below topics are covered in a finished form for the 19.50 release. Please review and let me know where things need to be tweaked.

- Interface
  - getAudiences
  - page
  - collect
  - setConsent
  - getConsent 
- Config
  - onTagReady
  - onProfileReady
  - audienceLocalStorage
  - autorun

### CODE REVIEW

File: lightning-tag/detailed-reference.md [hunk 1]
```diff
@@ -1,3 +1,185 @@
...
+onTagReady: function(namespace) {
+  // Lightning Tag script is ready to be used
+},
+```
+
+?> This event does not confirm audiences are available, that is the `onProfileReady` callback described above in this document.
+
+#### Auto Run Flag
+
+The `autoRun` boolean is defaulted `true` and runs the Lotame data collection and targeting retrieval is run immediately after the script is loaded. By default it is `true` so those functions run automatically after the script is loaded. If set to `false` control is left to the client's website to fire data collection and targeting using the `collect()` and `page()` methods described lower on this document.
```
Comment (singhald) [inline lightning-tag/detailed-reference.md]: ```The `autoRun` boolean is defaulted `true` and runs the Lotame data collection and targeting retrieval immediately after the script is loaded. If set to `false` control is left to the client's website to fire data collection and targeting using the `collect()` and `page()` methods described later on this document.```

changed some wording, there were artifacts of prior versions

File: lightning-tag/detailed-reference.md [hunk 2]
```diff
@@ -1,3 +1,185 @@
...
+
+### Data Object
+
+The `data` object is used to pass first party-data to the Lotame DMP for use in targeting or other features of the Lotame platform. The object is optionally passed at setup time in the tag input function. It can also be passed post-page load in the `collect({data})` or in the `page({data})` methods that is described further down this page.
+
+The `data` object's parameters are fully described in [Lightning Tag Data Collection](lightning-tag/data-collection.md).
+
+## Lightning Tag Methods
+
+!> To ensure any of the below methods are available, ensure the `onTagReady` callback has returned else the below methods are not available yet.
```
Comment (singhald) [inline lightning-tag/detailed-reference.md]: ```!> To use any of the below methods, ensure the `onTagReady` callback has returned else the below methods are not available yet.```

I don't like that "else ..."

```!> To ensure any of the below methods are available, ensure the `onTagReady` callback has returned``` ?

File: lightning-tag/detailed-reference.md [hunk 3]
```diff
@@ -1,3 +1,185 @@
...
+
+### collect()
+
+The `collect()` method is available to pass data to your Lotame DMP the Lotame javascript is loaded. An example use-case is tracking events such as user interaction with a video player on your site.
```
Comment (singhald) [inline lightning-tag/detailed-reference.md]: ... the Lotame DMP once the Lotame javascript ... ?

File: lightning-tag/detailed-reference.md [hunk 4]
```diff
@@ -1,3 +1,185 @@
...
+window.lotame_<lotameClientId>.getAudiences();
+```
+
+Audiences are written out as a string of comma-delimited targeting codes or audience IDs, based on your established tag settings. Example: `"aud1,aud2"` or `"123,789"`
+
+?> The `audienceLocalStorage` parameter must be set in the Lotame Lightning Tag `config` object. If it is not, the response to `getAudiences` will be an empty string and an error message written to `console` log.
+
+### setConsent()
+
+The Lotame Ligntning Tag provides a method `setConsent` method to submit consent signals to the Lotame Consent API for enforcement within the Lotame DMP. Once you've collected the appropriate consent from your customer within your privacy user experience, submit the appropriate signals as follows:
```
Comment (singhald) [inline lightning-tag/detailed-reference.md]: method setconsent method

**Review by singhald — CHANGES_REQUESTED** [review]

File: lightning-tag/detailed-reference.md (hunk 2 above)
Comment (ErikMHummel) [reply lightning-tag/detailed-reference.md]: 👍

File: lightning-tag/detailed-reference.md (hunk 4 above)
Comment (ErikMHummel) [reply lightning-tag/detailed-reference.md]: 👍

File: lightning-tag/detailed-reference.md (hunk 3 above)
Comment (ErikMHummel) [reply lightning-tag/detailed-reference.md]: I missed 'once'

File: lightning-tag/detailed-reference.md (hunk 1 above)
Comment (ErikMHummel) [reply lightning-tag/detailed-reference.md]: Much better.

**Review by singhald — APPROVED** [review]

**Pull request merged**